crates/storage/src/kv_store.rs

```diff
@@ -41,7 +44,7 @@ pub trait StorageColumn: Copy + core::fmt::Debug {
 
 /// The definition of the key-value inspection store.
 #[impl_tools::autoimpl(for<T: trait> &T, &mut T, Box<T>)]
-pub trait KeyValueInspect {
+pub trait KeyValueInspect: Send + Sync {
```

Comment:
It is very sad that we need to require `Send + Sync` only because `BoxedIter` requires `Send`. Maybe we can define non Send boxed iterator instead and use it(because it seems we don't need `Send` feature, but maybe I'm wrong)?

Reply:
Yeah this bound is just to satisfy the `BoxedIter` requirement. I can see if I can define a non-send one.

Reply:
Hitting some walls with this implementation. Will try again tomorrow with fresh eyes.

Reply:
Alright my mistake was trying to change some iterators that we turn into streams, and they need to be `Send`. Now I have managed to get rid of these trait bounds https://github.com/FuelLabs/fuel-core/pull/2396/commits/cbb6efc4332490122e8de70d082c7048cc97e0a4.

Now we have two boxed iterators: `BoxedIter` which doesn't require Send, and `BoxedIterSend` for the cases when you need Send.